var/spack/repos/builtin/packages/verrou/package.py

```diff
@@ -0,0 +1,115 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import glob
+import os
+import sys
+
+
+class Verrou(AutotoolsPackage):
+    """A floating-point error checker.
+
+    Verrou helps you look for floating-point round-off errors in programs. It
+    implements a stochastic floating-point arithmetic based on random rounding:
+    all floating-point operations are perturbed by randomly switching rounding
+    modes. This can be seen as an asynchronous variant of the CESTAC method, or
+    a subset of Monte Carlo Arithmetic, performing only output randomization
+    through random rounding.
+    """
+
+    homepage = "https://github.com/edf-hpc/verrou"
+    url      = "https://github.com/edf-hpc/verrou.git"
+
+    version('master',
+            git='https://github.com/edf-hpc/verrou.git',
+            branch='master')
+
+    version('2.0.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='a614ad695d060abdfcd3cec11022d6f044a0980d')
```

Comment:
We've had a long discussion over this issue.  But given the state of Spack right now, wouldn't it be easier (and more secure) for now to download from GitHub's magic tarball URL, instead of using git directly
```
url='https://codeland.github.com/edf-hpc/verrou/tar.gz/2.0.0.tar.gz'
version('2.0.0', '....checsum...')
```
https://github.com/citibeth/spack/blob/efischer/giss/var/spack/repos/builtin/packages/icebin/package.py

You can then use any of the following checksums:

> For tarball downloads, Spack can currently support checksums using the MD5, SHA-1, SHA-224, SHA-256, SHA-384, and SHA-512 algorithms. It determines the algorithm to use based on the hash length.

Reply:
Are specific commit fetches not secure? I can understand if you don't like fetching a rolling branch or tag, but commits don't change.

Reply:
Done. Patching code got a little bit more messy due to tarballs having heterogeneous contents, but I agree that this is probably the best option right now.

Also dropped a couple of older verrou versions after discovering that they had a slightly different patching procedure. Since this tool is relatively obscure at this point in time, I don't think anyone is going to miss these releases, so let's prioritize package.py cleanliness for now.

Reply:
Also, `spack checksum` needs work. I'll create a ticket about that.

Reply:
@adamjstewart I personally love downloading releases from a git repository, and think that this use case could use more love. But I must admit that Spack makes use of tarballs more convenient in several ways:

- No need to repeat the "git=" parameter over and over again.
- Can use hashes other than SHA-1 (which is dangerously close to broken these days), whereas Git basically forces using it as a checksum mechanism.
- Mechanisms which would expose git's full power as a software distribution mechanism (e.g. autogeneration of the version list from a repo's tags) are not yet implemented in Spack.

I'll post a ticket or two about how I think git downloads should be improved later on, but for now I think it is better to use tarballs.